### PR TITLE
Feature/112 auth pass change

### LINKED ISSUE #112: feature: Password Change (Authenticated)
### Summary

Implement password change endpoint and frontend UI to allow authenticated users to change their password from the settings screen, requiring current password verification.

---

### Description

Authenticated users should be able to change their password from the settings screen. This feature requires the user to provide their current password for security verification, then allows them to set a new password. This is different from password reset (forgot password) as it requires authentication and current password verification.

---

### User Flow

**Primary Flow:**
1. User is logged in and navigates to Settings screen
2. User taps "Change Password" option
3. User navigates to Change Password screen
4. User enters current password
5. User enters new password
6. User confirms new password
7. User sees password requirements and real-time validation
8. User submits password change request
9. System verifies current password and updates to new password
10. User sees success message
11. User can continue using app with new password

**Alternative Flows:**
- User cancels: Returns to settings screen without changes
- User forgets current password: User must use forgot password flow instead

**Error Scenarios:**
- Invalid current password: Show error message, allow retry
- Password mismatch: Show error message
- Password doesn't meet requirements: Show error message
- Network error: Show error message, allow retry
- Unauthorized (token expired): Log user out, redirect to login
- Rate limiting: Show error message about too many requests

---

#### Endpoints to Implement

**Endpoint 1: `POST /password/change`**
- **Operation ID:** `changePassword`
- **Description:** Changes the authenticated user's password. Requires a valid access token.
- **Security:** Requires Bearer token authentication
- **Request Body:**
  - `currentPassword` (string, required): User's current password
  - `newPassword` (string, required): New password (must meet Cognito password policy)
  - `newPasswordConfirm` (string, required): Password confirmation (must match newPassword)
- **Success Response (200):**
  - Schema: Object with `message` property
  - Example: `{ "message": "Password changed successfully" }`
- **Error Responses:**
  - `400`: Bad request - password mismatch, invalid current password, or missing fields
  - `401`: Unauthorized - invalid or missing access token
  - `429`: Too many requests - rate limit exceeded
  - `500`: Internal server error

#### Backend Requirements

- [ ] Implement Lambda handler function for endpoint
- [ ] Integrate with AWS Cognito using `ChangePassword` API
- [ ] Require Bearer token authentication (extract from Authorization header)
- [ ] Handle error cases and sanitize error messages
- [ ] Add input validation and normalization
- [ ] Validate password requirements match Cognito policy
- [ ] Validate password confirmation matches new password
- [ ] Verify current password is correct
- [ ] Add unit tests for handler logic
- [ ] Add integration tests for endpoint
- [ ] Update CDK stack if new Lambda function or route needed

#### Cognito APIs Used

- `ChangePassword`: Change authenticated user's password (requires access token)

---

#### Screens/Components to Create/Modify

**Screen: `ChangePasswordScreen`**
- **Location:** `frontend/app/(tabs)/change-password.tsx` or `frontend/app/(auth)/change-password.tsx`
- **Purpose:** Screen for changing password
- **Key Features:**
  - Text input for current password (secure entry)
  - Text input for new password (secure entry with show/hide toggle)
  - Text input for password confirmation (secure entry with show/hide toggle)
  - Password requirements display
  - Real-time password matching feedback
  - Submit button to change password
  - Cancel button to return to settings
  - Success message and navigation back to settings
  - Error handling and display
- **User Interactions:**
  - User enters current password
  - User enters new password
  - User confirms password
  - User submits change request
  - User sees success and returns to settings

**Screen: `SettingsScreen` (Modify)**
- **Location:** `frontend/app/(tabs)/settings.tsx`
- **Purpose:** Add "Change Password" option to settings
- **Key Features:**
  - Add "Change Password" button/option
  - Navigate to change password screen
- **User Interactions:**
  - User taps "Change Password"
  - User navigates to change password screen

**Component: `PasswordRequirements` (Reuse)**
- **Location:** Reuse from registration flow
- **Purpose:** Display password requirements
- **Key Features:**
  - Show password requirements checklist
  - Real-time validation feedback

#### Frontend Requirements

- [ ] Create change password screen component
- [ ] Add "Change Password" option to settings screen
- [ ] Implement API service function for endpoint
- [ ] Add form validation and error handling
- [ ] Implement loading states and user feedback
- [ ] Add navigation between screens
- [ ] Ensure accessibility (labels, hints, roles)
- [ ] Follow existing design system (colors, spacing, typography)
- [ ] Handle edge cases (network errors, timeouts, token expiration)
- [ ] Add unit/integration tests for components
- [ ] Reuse password requirements component from registration
- [ ] Include access token in API request headers

#### API Service Functions

- `changePassword`: Change authenticated user's password
  - Parameters: `currentPassword`, `newPassword`, `newPasswordConfirm`
  - Returns: `Promise<{ message: string }>`
  - Error handling: Display user-friendly error messages, handle password mismatch, handle 401 by logging out

---

### OpenAPI Specification

- [ ] Verify endpoint is documented in `docs/openapi.yaml`
- [ ] Update OpenAPI spec if schema changes needed
- [ ] Ensure request/response schemas are defined
- [ ] Add/update examples for endpoint
- [ ] Verify BearerAuth security scheme is defined

**Endpoints in OpenAPI:**
- `POST /password/change` - `changePassword`

---

### Acceptance Criteria

**Backend:**
- [ ] Endpoint returns correct status codes and responses
- [ ] Endpoint requires valid Bearer token authentication
- [ ] Error handling works correctly for all error scenarios
- [ ] Input validation prevents invalid requests
- [ ] Password validation matches Cognito requirements
- [ ] Current password verification works correctly
- [ ] Integration tests pass for endpoint
- [ ] Unit tests have adequate coverage
- [ ] Error messages are sanitized and user-friendly

**Frontend:**
- [ ] User can navigate to change password screen from settings
- [ ] User can change password successfully
- [ ] Current password verification works correctly
- [ ] Password requirements are displayed and validated
- [ ] Password matching is validated in real-time
- [ ] Error scenarios are handled gracefully with user-friendly messages
- [ ] Loading states are shown during API calls
- [ ] Form validation provides real-time feedback
- [ ] Navigation works correctly between screens
- [ ] UI follows design system and accessibility guidelines
- [ ] Components are properly tested
- [ ] Access token is included in API requests

**Integration:**
- [ ] Frontend successfully calls backend endpoint with authentication
- [ ] End-to-end flow works from settings to password change completion
- [ ] Error messages are displayed correctly to users
- [ ] User can continue using app with new password after change
- [ ] 401 errors properly log user out

---

### Dependencies

**Prerequisites:**
- [ ] Login endpoint (`/login`) must be implemented
- [ ] Settings screen must exist
- [ ] Token storage and authentication context must exist
- [ ] Bearer token authentication must be working

**Blocks:**
- [ ] None identified

---

### Testing Requirements

**Unit Tests:**
- [ ] Backend handler logic tests
- [ ] Frontend component tests
- [ ] API service function tests
- [ ] Password validation logic tests
- [ ] Authentication token handling tests

**Integration Tests:**
- [ ] Backend endpoint integration tests (with authentication)
- [ ] Frontend-backend integration tests
- [ ] Test password change flow end-to-end with authentication

**Manual Testing:**
- [ ] Test change password flow from settings
- [ ] Test current password verification
- [ ] Test all error scenarios
- [ ] Test password requirements validation
- [ ] Test password matching validation
- [ ] Test authentication token handling
- [ ] Test 401 error handling (token expiration)
- [ ] Test on iOS and Android devices
- [ ] Test with various input formats/edge cases

---

### Technical Notes

- This endpoint requires authentication (Bearer token)
- Current password must be verified before allowing change
- Password requirements must match Cognito policy
- User should be able to continue using app immediately after password change
- Consider logging password change events for security auditing
- On 401 error (token expired), user should be logged out and redirected to login
- Access token should be extracted from Authorization header: `Bearer <token>`

---

### Related Issues

- Related to: Settings screen implementation
- Related to: Login flow implementation
- Part of: Authentication system

### PR DESCRIPTION
**Pull request by rtipper1**
## Summary

Implements authenticated password change in the frontend settings flow and aligns password validation UI with registration.

- Refactors settings tab into a nested stack (`settings/_layout`, `settings/index`, `settings/change-password`) and adds navigation to the new change-password screen.
- Adds `changePassword` API support with bearer access token handling, clear 401/session-expired messaging, and screen-level logout on expired sessions.
- Extracts shared password policy logic into `frontend/utils/passwordPolicy.ts` and reuses it in both registration and change-password flows.
- Standardizes password requirement and password-match indicators to consistent icon-based UI (green checkmarks, outlined/unmatched states) across flows.
- Adds tests for `api.changePassword` and the new change-password screen behavior (validation, success path, and session-expired handling).

closes #112

### CODE REVIEW

File: aws_resources/backend/src/main/kotlin/com/handlers/AuthHandler.kt [hunk 1]
```diff
@@ -565,6 +567,126 @@ class AuthHandler : RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyR
         }
     }
 
+    private fun handleChangePassword(
```
Comment (kdigava) [inline aws_resources/backend/src/main/kotlin/com/handlers/AuthHandler.kt:694]: The backend validation checking if newPassword == currentPassword is a good safety catch. However, are we relying solely on the Cognito User Pool's password policy for complexity (length, special chars, etc.), or should we implement a secondary check here to provide more specific error messages before hitting the AWS limit?"

File: aws_resources/backend/src/main/kotlin/com/handlers/AuthHandler.kt [hunk 2]
```diff
@@ -45,6 +46,7 @@ class AuthHandler : RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyR
         val response = when {
             path == "/login" && method == "POST" -> handleLogin(input, context)
             path == "/register" && method == "POST" -> handleRegister(input, context)
```
Comment (kdigava) [inline aws_resources/backend/src/main/kotlin/com/handlers/AuthHandler.kt:50]: Since we are introducing the /password/change endpoint, should we verify that we’ve updated the CORS configuration in the AuthHandler or the API Gateway settings to allow this specific path for the frontend origin."

File: aws_resources/backend/src/main/kotlin/com/handlers/AuthHandler.kt (hunk 1 above)
Comment (rtipper1) [reply aws_resources/backend/src/main/kotlin/com/handlers/AuthHandler.kt:694]: The frontend does not allow the user to submit the password change form unless it passes the same criteria so relying on the Cognito User Pool's password policy should be fine.

File: aws_resources/backend/src/main/kotlin/com/handlers/AuthHandler.kt (hunk 2 above)
Comment (rtipper1) [reply aws_resources/backend/src/main/kotlin/com/handlers/AuthHandler.kt:50]: CORS should not be an issue when accessing the endpoint in our production environment. Android/iOS app / expo app. This could be an issue when manually testing the endpoint, I will make note of this and come back to it if we have issues with that.

**Review by AkashK321 — APPROVED** [review]
